### ISSUE TITLE
Bazel build fails : error loading package 'syntaxnet': Package 'syntaxnet' contains errors

### ISSUE THREAD

**Issue by SaintNazaire** [tensorflow/models#3047]
**System information**

- Windows 10 professional
- Docker image based on Ubuntu 16.04.3 LTS
- Docker 17.09.0-ce
- Python version 2.7
- Bazel version 0.7.0 AND 0.5.4
- CUDA/cuDNN version not relevant (CPU install)
- GPU model and memory not relevant (CPU install)

**Command triggering issue**

`cd /models/research/syntaxnet && test ... --genrule_strategy=standalone --spawn_strategy=standalone
`

**Top-level directory of the model you are using**

> /models/research/syntaxnet

**Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**

Well tried and tested docker build files under two months old, the standard issue with Bazel 0.5.4 following exact install procedure and then one with Bazel 0.7.0 and a little work-around to disable dynamically declared static variables (cf. of issue tensorflow/tensorflow#12491 and #2355)

Only variation I made was to clearly specify environment variables:

```
ENV PYTHON_BIN_PATH=/usr/bin/python
ENV PYTHON_LIB_PATH=/usr/lib/python2.7
```

In addition added sandboxing exception handling:

`cd /models/research/syntaxnet && bazel test ... --genrule_strategy=standalone --spawn_strategy=standalone`

**Describe the problem**

Tried and tested docker build files fail to load dragnn/protos, returns error, despite several attempts checks, internet connection reset, machine reboot, memory and cpu alloted to docker changes.

> error loading package 'syntaxnet': Package 'syntaxnet' contains errors

**Error logs**

> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:35:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 35
>                 tf_proto_library_py(name = "data_py_pb2", srcs = ["dat..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:40:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 40
>                 tf_proto_library_py(name = "trace_py_pb2", srcs = ["tr..."], ..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:46:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 46
>                 tf_proto_library_py(name = "spec_py_pb2", srcs = ["spe..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: package contains errors: dragnn/protos
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:13:1: First argument of 'load' must be a label and start with either '//', ':', or '@'. Use --incompatible_load_argument_is_label=false to temporarily disable this check.
> ERROR: package contains errors: syntaxnet
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:23:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:28:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:33:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:38:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:43:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:48:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:54:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:59:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:64:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:69:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:773:1: name 'tf_gen_op_libs' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:777:1: name 'tf_gen_op_wrapper_py' is not defined
> ERROR: error loading package 'syntaxnet': Package 'syntaxnet' contains errors
> INFO: Elapsed time: 43.547s
> FAILED: Build did NOT complete successfully (19 packages loaded)

**Early diagnostic**

Possible link with issue #2822, #2962 i.e. Bazel

**Suggestion**

Syntaxnet is almost always down those days due to one problem or another.

**Comment by jart** [comment]
I just approved #2962 which might help get this unblocked. I'm going to assume this issue is a duplicate of #2822. I would redirect further concerns to that issue. I'll close this one out since it's either a duplicate or subset.

**Comment by shubham0704** [comment]
I am facing the same issue  even after the fix of bazel version to latest - 
![screenshot from 2018-01-07 12-49-26](https://user-images.githubusercontent.com/15546524/34647498-2d3208de-f3aa-11e7-8580-c4aea8c4b570.png)

Possibly buggy implementation of py_proto_library - 
https://stackoverflow.com/questions/47688252/tensorflow-trying-to-mutate-a-frozen-object-bazel

**Comment by SaintNazaire** [comment]
Suggest to reopen the issue problem is still there, the error log is almost identical.

Bazel version 0.8.0 AND 0.7.0 AND 0.5.4

> Loading:
> Loading: 0 packages loaded
> Loading: 0 packages loaded
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> Loading: 17 packages loaded
>     currently loading: dragnn/protos ... (2 packages)
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:35:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 35
>                 tf_proto_library_py(name = "data_py_pb2", srcs = ["dat..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:40:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 40
>                 tf_proto_library_py(name = "trace_py_pb2", srcs = ["tr..."], ..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: /models/research/syntaxnet/dragnn/protos/BUILD:46:1: Traceback (most recent call last):
>         File "/models/research/syntaxnet/dragnn/protos/BUILD", line 46
>                 tf_proto_library_py(name = "spec_py_pb2", srcs = ["spe..."])
>         File "/models/research/syntaxnet/syntaxnet/syntaxnet.bzl", line 53, in tf_proto_library_py
>                 py_proto_library(name = name, srcs = srcs, srcs_versi...", <5 more arguments>)
>         File "/root/.cache/bazel/_bazel_root/3b4c7ccb85580bc382ce4a52e9580003/external/protobuf_archive/protobuf.bzl", line 374, in py_proto_library
>                 py_libs += [default_runtime]
> trying to mutate a frozen object
> ERROR: package contains errors: dragnn/protos
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:13:1: First argument of 'load' must be a label and start with either '//', ':', or '@'. Use --incompatible_load_argument_is_label=false to temporarily disable this check.
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:23:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:28:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:33:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:38:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:43:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:48:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:54:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:59:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:64:1: name 'tf_proto_library' is not defined (did you mean 'cc_proto_library'?)
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:69:1: name 'tf_proto_library_py' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:773:1: name 'tf_gen_op_libs' is not defined
> ERROR: /models/research/syntaxnet/syntaxnet/BUILD:777:1: name 'tf_gen_op_wrapper_py' is not defined
> ERROR: package contains errors: syntaxnet
> ERROR: error loading package 'syntaxnet': Package 'syntaxnet' contains errors
> INFO: Elapsed time: 13.186s
> FAILED: Build did NOT complete successfully (19 packages loaded)
